zap/src/main/java/org/zaproxy/zap/extension/script/ScriptNode.java

```diff
@@ -97,4 +98,17 @@ public ScriptType getType() {
     public boolean isTemplate() {
         return template;
     }
+
+    @Override
+    public ScriptWrapper getUserObject() {
+        return script;
+    }
+
+    @Override
+    public void setUserObject(Object userObject) {
+        if (!(userObject instanceof ScriptWrapper)) {
+            throw new IllegalArgumentException("Parameter userObject must be a ScriptWrapper.");
+        }
+        this.script = (ScriptWrapper) userObject;
```

Comment:
You should be able to assign the name here (at instanceof) and not cast below

Reply:
I considered that, but I felt this was more readable...